### PR TITLE
MULE-13065: Privileged API packages must take into account other expo…

### PR DESCRIPTION
**Pull request by pablokraan**
…rted privileged APIs

_ Taking in account privileged packages exported by module dependencies
_ Refactoring code

### CODE REVIEW

File: src/main/java/org/mule/tools/maven/plugin/module/analyze/Module.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+ */
+
+package org.mule.tools.maven.plugin.module.analyze;
+
+import java.util.Set;
+
+/**
+ * Contains all the required information about a given Mule module that is needed for the API analysis.
+  *
+  * @since 1.0
```
Comment (elrodro83) [inline src/main/java/org/mule/tools/maven/plugin/module/analyze/Module.java]: format

File: src/main/java/org/mule/tools/maven/plugin/module/analyze/ModuleDiscoverer.java [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+      elements = project.getRuntimeClasspathElements();
+      elements.addAll(project.getCompileClasspathElements());
+
+      for (String element : elements) {
+        urls.add(new File(element).toURI().toURL());
+      }
+
+      ClassLoader contextClassLoader = URLClassLoader.newInstance(
+                                                                  urls.toArray(new URL[0]),
+                                                                  Thread.currentThread().getContextClassLoader());
```
Comment (elrodro83) [inline src/main/java/org/mule/tools/maven/plugin/module/analyze/ModuleDiscoverer.java]: static imports

File: src/main/java/org/mule/tools/maven/plugin/module/analyze/ModuleDiscoverer.java [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+      try {
+        final Enumeration<URL> resources = contextClassLoader.getResources("META-INF/mule-module.properties");
```
Comment (elrodro83) [inline src/main/java/org/mule/tools/maven/plugin/module/analyze/ModuleDiscoverer.java]: extract constants

File: src/main/java/org/mule/tools/maven/plugin/module/analyze/ModuleDiscoverer.java [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+  private Properties getModuleProperties(MavenProject project) throws ModuleApiAnalyzerException {
+    Properties properties = null;
+    try {
+      final List<Resource> projectResources = project.getBuild().getResources();
+      File result = null;
+      for (int i = 0; i < projectResources.size(); i++) {
+        final Resource resource = projectResources.get(i);
+
+        File moduleProperties1 = new File(resource.getDirectory(), "META-INF" + File.separator + "mule-module.properties");
```
Comment (elrodro83) [inline src/main/java/org/mule/tools/maven/plugin/module/analyze/ModuleDiscoverer.java]: extract constants

Comment (elrodro83) [inline src/main/java/org/mule/tools/maven/plugin/module/analyze/ModuleDiscoverer.java]: remove the `1`

File: src/main/java/org/mule/tools/maven/plugin/module/analyze/ModuleDiscoverer.java [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+
+      ClassLoader contextClassLoader = URLClassLoader.newInstance(
+                                                                  urls.toArray(new URL[0]),
+                                                                  Thread.currentThread().getContextClassLoader());
+
+      try {
+        final Enumeration<URL> resources = contextClassLoader.getResources("META-INF/mule-module.properties");
+        while (resources.hasMoreElements()) {
+          final URL url = resources.nextElement();
+          Properties properties = loadProperties(url);
```
Comment (elrodro83) [inline src/main/java/org/mule/tools/maven/plugin/module/analyze/ModuleDiscoverer.java]: can this be refactored to use the uri instead of the url?

Comment (pablokraan) [inline src/main/java/org/mule/tools/maven/plugin/module/analyze/ModuleDiscoverer.java]: Classloader returns URLs and I will need to open a connection to load the file, so it must use URLs anyway

File: src/main/java/org/mule/tools/maven/plugin/module/analyze/ModuleDiscoverer.java [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+   */
+  public Module discoverProjectModule(MavenProject project, AnalyzerLogger analyzerLogger)
+      throws ModuleApiAnalyzerException {
+    Module module = null;
+    Properties properties = getModuleProperties(project);
+    if (properties != null) {
+      try {
+        module = moduleFactory.create(analyzerLogger, (String) properties.get("module.name"), properties);
+      } catch (IOException e) {
+        throw new ModuleApiAnalyzerException("Cannot read project's mule-module.properties", e);
```
Comment (elrodro83) [inline src/main/java/org/mule/tools/maven/plugin/module/analyze/ModuleDiscoverer.java]: extract `mule-module.properties` to a constant too

**Pull request merged**